composer/utils/checkpoint.py

```diff
@@ -276,9 +291,17 @@ def read_data(self, plan: LoadPlan, planner: LoadPlanner):
                 # PyTorch will capture any exception of this function,
                 # and dist.all_gather_objects(exception) before raising it.
                 # If that all_gather_objects fails, the exception is never visible to user.
-                # We immediately print the exception to avoid that situation.
+                # We immediately kill the process and print the exception
                 log.error(f'Exception {type(e)} raised during downloading: {str(e)}')
-                raise e
+                download_error = True
+
+        download_error_tensor = torch.tensor(1 if download_error else 0)
+        dist.all_reduce(download_error_tensor, reduce_operation='SUM')
+        if download_error_tensor.item() > 0:
+            self.terminate_all_processes()
+            log.error(
+                f'{download_error_tensor.item()} downloads failed across all nodes. Terminating all processes due to download error.'
+            )
```

Comment:
Can you broadcast which rank failed and say look at rank X to see the exception (which is logged as `log.error`)?